web/twa-vis-platform/code/src/ui/map/mapbox/mapbox-container.tsx

```diff
@@ -29,67 +29,84 @@ interface MapProperties {
-  const mapRef = useRef();
-  const mapContainerRef = useRef();
+    const mapContainerRef = useRef();
 
-  // Run when component loaded
-  useEffect(() => {
-    initialiseMap();
+    // Run when component loaded
+    useEffect(() => {
+        initialiseMap();
 
-    return () => {
-      if (props.currentMap) {
-        props.currentMap.remove(); // Remove the map instance
-        props.setMap(null); // Reset the map ref
-      }
-    };
-  }, []);
+        return () => {
+            if (props.currentMap) {
+                props.currentMap.remove(); // Remove the map instance
+                props.setMap(null); // Reset the map ref
+            }
+        };
+    }, []);
 
-  // Initialise the map object
-  const initialiseMap = async () => {
-    props.currentMap?.remove();
+    // Initialise the map object
+    const initialiseMap = async () => {
+        props.currentMap?.remove();
 
-    const defaultImagery: ImageryOption = props.imageryOption ?? {
-      "name": "Standard (Night)",
-      "url": "mapbox://styles/mapbox/standard",
-      "time": "dusk"
-    };
+        const defaultImagery: ImageryOption = props.imageryOption ?? {
+            "name": "Standard (Night)",
+            "url": "mapbox://styles/mapbox/standard",
+            "time": "dusk"
+        };
 
-    mapboxgl.accessToken = process.env.MAPBOX_API_KEY;
+        mapboxgl.accessToken = process.env.MAPBOX_API_KEY;
 
-    const map: Map = new mapboxgl.Map({
-      container: mapContainerRef.current,
-      style: defaultImagery.url,
-      center: props.defaultPosition.center,
-      zoom: props.defaultPosition.zoom,
-      bearing: props.defaultPosition.bearing,
-      pitch: props.defaultPosition.pitch,
-    });
+        const map: Map = new mapboxgl.Map({
+            container: mapContainerRef.current,
+            style: defaultImagery.url,
+            center: props.defaultPosition.center,
+            zoom: props.defaultPosition.zoom,
+            bearing: props.defaultPosition.bearing,
+            pitch: props.defaultPosition.pitch,
+            transformRequest: (url) => {
+                if (process.env.REACT_APP_USE_GEOSERVER_PROXY === 'true') {
+                    try {
+                        const urlObject = new URL(url);
+                        const params = new URLSearchParams(urlObject.search);
+                        if (params.get('request') === 'GetMap') {
+                            // not sure if this will work across all conditions
+                            const serverUrl = `${window.location.protocol}//${window.location.host}`;
+                            const proxyUrl = `${serverUrl}/geoserver-proxy?url=${encodeURIComponent(url)}`;
+                            return {
+                                url: proxyUrl
+                            }
+                        }
+                    } catch { }
+                } else {
+                    return { url: url }
+                }
+            }
```

Comment:
```suggestion
            transformRequest: (url, resourceType) => {
        if (resourceType === 'Source' && url.startsWith('http://myHost')) {
            return {
                url: url.replace('http', 'https'),
                headers: {'my-custom-header': true},
                credentials: 'include'  // Include cookies for cross-origin requests
            };
        }
    }
```

comes from [here](https://docs.mapbox.com/mapbox-gl-js/api/map/). Let's try just pass the header in this bit of code and use a variable to get the bearer token from /api/userinfo

Reply:
Using resourceType does not seem to be robust enough, when I use MapBox vector tile, its value is 'Tile'